api/src/user/controllers/role.controller.ts

```diff
@@ -142,12 +147,24 @@ export class RoleController extends BaseController<Role, RoleStub> {
   @CsrfCheck(true)
   @Delete(':id')
   @HttpCode(204)
-  async deleteOne(@Param('id') id: string) {
-    const result = await this.roleService.deleteOne(id);
-    if (result.deletedCount === 0) {
-      this.logger.warn(`Unable to delete Role by id ${id}`);
-      throw new NotFoundException(`Role with ID ${id} not found`);
+  async deleteOne(@Param('id') id: string, @Session() session: ExpressSession) {
+    const roles = (
+      await this.userService.findOneAndPopulate(session.passport?.user?.id, [
```

Comment:
```suggestion
      await this.userService.findOneAndPopulate(req.user, [
```
I think the id can be found in req, can you check ?

Reply:
Attempting to access req.user.id results in an error. Instead, the user ID can be accessed via req.session.passport.user.id or session directly like i wrote in my code.

![image](https://github.com/user-attachments/assets/90da8c39-7f41-4d25-9e28-5ef3d5940072)

Reply:
it's only a typecheck error, the actual value is there.
You can easily solve that by a simple cast to user but that's not how you should do this in this case.
The user object is already there so there is no need to fetch it again with findOneAndPopulate, You only need to fetch the roles.

But seeing that you didn't use the roles afterwards, you only used their IDs, there is no need to even to fetch the roles, you already have a roles array in the object user containing role ids

Similarly if you did have to fetch the user ( which is not the case), you wouldn't need to populate the roles since you only need the roles ids

Reply:
Actually the user exists but with an id, first_name and last_name attributes as shown in the following screenshot, so i need to fetch the user data with findOneAndPopulate
![image](https://github.com/user-attachments/assets/df700193-ee66-41db-a9eb-d3d1c612c085)